**Hello, welcome to the preprocessing notebook of our data science project: predicting successful startup!**

We will try to include as much details as possible with markdowns :)

# Data Import

In [78]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [79]:
# data = pd.read_csv('/Users/danyhersco/data_startup_final.csv')
# this is mine!!!

# Data Exploration

In [80]:
data.shape

(314486, 19)

Let's quickly drop real duplicates (see below for "false duplicates"):

In [84]:
data = data.drop_duplicates()
data.shape

(313349, 19)

We extracted the dataset from DBeaver, after running an SQL query. As a result of multiple `LEFT JOIN` to include `industry` and `technology` features, many startups have duplicates.

Consequently, let's check for the number of unique values for each feature of our dataframe.

In [85]:
data.describe(include='all')

name                   website  \
count                313349                    310151   
unique                55665                     54614   
top     Arthur Intelligence  https://www.goarthur.ai/   
freq                    216                       216   
mean                    NaN                       NaN   
std                     NaN                       NaN   
min                     NaN                       NaN   
25%                     NaN                       NaN   
50%                     NaN                       NaN   
75%                     NaN                       NaN   
max                     NaN                       NaN   

                    short_description ipo_status  founded_on went_public_on  \
count                          313346     313349      313349           5368   
unique                          56142          3        1083            507   
top     Limited information available    private  2016-01-01     2021-11-11   
freq                              597     308312       67276            114   
mean                              NaN        NaN         NaN            NaN   
std                               NaN        NaN         NaN            NaN   
min                               NaN        NaN         NaN            NaN   
25%                               NaN        NaN         NaN            NaN   
50%                               NaN        NaN         NaN            NaN   
75%                               NaN        NaN         NaN            NaN   
max                               NaN        NaN         NaN            NaN   

         exited_on  num_funding_rounds last_equity_funding_type  \
count        22027       313349.000000                   287895   
unique        1370                 NaN                       18   
top     2021-02-19                 NaN                     seed   
freq           146                 NaN                   107263   
mean           NaN            3.410079                      NaN   
std            NaN            2.420720                      NaN   
min            NaN            1.000000                      NaN   
25%            NaN            2.000000                      NaN   
50%            NaN            3.000000                      NaN   
75%            NaN            5.000000                      NaN   
max            NaN           22.000000                      NaN   

                                last_equity_funding_total last_funding_at  \
count                                              235682          313349   
unique                                              15347            2905   
top     {"currency":"USD","amount":100000000,"amountUS...      2019-01-01   
freq                                                 4776            2882   
mean                                                  NaN             NaN   
std                                                   NaN             NaN   
min                                                   NaN             NaN   
25%                                                   NaN             NaN   
50%                                                   NaN             NaN   
75%                                                   NaN             NaN   
max                                                   NaN             NaN   

       headquartersCountry headquartersRegion  employeeCount industry_name  \
count               312208             312165  301115.000000        304228   
unique                 162               1392            NaN            40   
top                     US         California            NaN   Health Care   
freq                138007              49580            NaN         29285   
mean                   NaN                NaN      61.976743           NaN   
std                    NaN                NaN     443.794835           NaN   
min                    NaN                NaN       0.000000           NaN   
25%                    NaN         

We have **55665** unique startups in our dataset. The next step will be to remove all duplicates while treating the non-duplicated columns.

Here is how our dataset look like:

In [86]:
data.head()

name                website  \
0  Cardiomo Care, Inc.  https://cardiomo.com/   
1  Cardiomo Care, Inc.  https://cardiomo.com/   
2  Cardiomo Care, Inc.  https://cardiomo.com/   
3  Cardiomo Care, Inc.  https://cardiomo.com/   
4  Cardiomo Care, Inc.  https://cardiomo.com/   

                                   short_description ipo_status  founded_on  \
0  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
1  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
2  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
3  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
4  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   

  went_public_on exited_on  num_funding_rounds last_equity_funding_type  \
0            NaN       NaN                  10                     seed   
1            NaN       NaN                  10                     seed   
2            NaN       NaN                  10                     seed   
3            NaN       NaN                  10                     seed   
4            NaN       NaN                  10                     seed   

                           last_equity_funding_total last_funding_at  \
0  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
1  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
2  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
3  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
4  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   

  headquartersCountry headquartersRegion  employeeCount industry_name  \
0                  US           New York           31.0   Health Care   
1                  US           New York           31.0   Health Care   
2                  US           New York           31.0   Health Care   
3                  US           New York           31.0   Health Care   
4                  US           New York           31.0   Health Care   

  technology_name              announcedOn     stage  \
0        Software  2019-04-15 00:00:00.000   unknown   
1        Software  2020-08-01 00:00:00.000      seed   
2        Software  2017-04-01 00:00:00.000   unknown   
3        Software  2016-08-09 00:00:00.000  pre_seed   
4        Software  2019-02-01 00:00:00.000  pre_seed   

                                         moneyRaised  
0  {"amount": 5652200, "currency": "USD", "amount...  
1  {"amount": 10000000, "currency": "USD", "amoun...  
2  {"amount": 2500000, "currency": "USD", "amount...  
3  {"amount": 13000000, "currency": "USD", "amoun...  
4  {"amount": 12000000, "currency": "USD", "amoun...

# Remove the duplicates through encoding

This problem will be challenging as there is not really fully duplicated rows. As we said above, it is just a result of the `LEFT JOIN` that multiplicated the rows for one company (several industries, several technologies, and several funding rounds). We have to one hot encode the latter features, which is what we will be doing in this section.

At the end of this section, we should reach a shape of `(55665, >18)`.

In [27]:
from useful.variables import industries, technologies

## `industry`

In [28]:
data.reset_index(drop=True, inplace=True)

In [29]:
data.head()

name                website  \
0  Cardiomo Care, Inc.  https://cardiomo.com/   
1  Cardiomo Care, Inc.  https://cardiomo.com/   
2  Cardiomo Care, Inc.  https://cardiomo.com/   
3  Cardiomo Care, Inc.  https://cardiomo.com/   
4  Cardiomo Care, Inc.  https://cardiomo.com/   

                                   short_description ipo_status  founded_on  \
0  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
1  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
2  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
3  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
4  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   

  went_public_on exited_on  num_funding_rounds last_equity_funding_type  \
0            NaN       NaN                  10                     seed   
1            NaN       NaN                  10                     seed   
2            NaN       NaN                  10                     seed   
3            NaN       NaN                  10                     seed   
4            NaN       NaN                  10                     seed   

                           last_equity_funding_total last_funding_at  \
0  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
1  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
2  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
3  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
4  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   

  headquartersCountry headquartersRegion  employeeCount industry_name  \
0                  US           New York           31.0   Health Care   
1                  US           New York           31.0   Health Care   
2                  US           New York           31.0   Health Care   
3                  US           New York           31.0   Health Care   
4                  US           New York           31.0   Health Care   

  technology_name              announcedOn     stage  \
0        Software  2019-04-15 00:00:00.000   unknown   
1        Software  2020-08-01 00:00:00.000      seed   
2        Software  2017-04-01 00:00:00.000   unknown   
3        Software  2016-08-09 00:00:00.000  pre_seed   
4        Software  2019-02-01 00:00:00.000  pre_seed   

                                         moneyRaised  
0  {"amount": 5652200, "currency": "USD", "amount...  
1  {"amount": 10000000, "currency": "USD", "amoun...  
2  {"amount": 2500000, "currency": "USD", "amount...  
3  {"amount": 13000000, "currency": "USD", "amoun...  
4  {"amount": 12000000, "currency": "USD", "amoun...

In [10]:
from sklearn.preprocessing import OneHotEncoder

encoder = OneHotEncoder(sparse=False)
data[encoder.get_feature_names_out()] = encoder.fit_transform(data[['industry_name']])

In [12]:
data.head()

name                website  \
0  Cardiomo Care, Inc.  https://cardiomo.com/   
1  Cardiomo Care, Inc.  https://cardiomo.com/   
2  Cardiomo Care, Inc.  https://cardiomo.com/   
3  Cardiomo Care, Inc.  https://cardiomo.com/   
4  Cardiomo Care, Inc.  https://cardiomo.com/   

                                   short_description ipo_status  founded_on  \
0  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
1  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
2  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
3  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
4  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   

  went_public_on exited_on  num_funding_rounds last_equity_funding_type  \
0            NaN       NaN                  10                     seed   
1            NaN       NaN                  10                     seed   
2            NaN       NaN                  10                     seed   
3            NaN       NaN                  10                     seed   
4            NaN       NaN                  10                     seed   

                           last_equity_funding_total  ...  \
0  {"currency":"USD","amount":10000000,"amountUSD...  ...   
1  {"currency":"USD","amount":10000000,"amountUSD...  ...   
2  {"currency":"USD","amount":10000000,"amountUSD...  ...   
3  {"currency":"USD","amount":10000000,"amountUSD...  ...   
4  {"currency":"USD","amount":10000000,"amountUSD...  ...   

  industry_name_Privacy and Security industry_name_Professional Services  \
0                                0.0                                 0.0   
1                                0.0                                 0.0   
2                                0.0                                 0.0   
3                                0.0                                 0.0   
4                                0.0                                 0.0   

  industry_name_Real Estate and Construction  \
0                                        0.0   
1                                        0.0   
2                                        0.0   
3                                        0.0   
4                                        0.0   

   industry_name_Sales and Marketing industry_name_Software  \
0                                0.0                    0.0   
1                                0.0                    0.0   
2                                0.0                    0.0   
3                                0.0                    0.0   
4                                0.0                    0.0   

  industry_name_Sports industry_name_Transportation  \
0                  0.0                          0.0   
1                  0.0                          0.0   
2                  0.0                          0.0   
3                  0.0                          0.0   
4                  0.0                          0.0   

  industry_name_Travel and Tourism industry_name_Video  industry_name_nan  
0                              0.0                 0.0                0.0  
1                              0.0                 0.0                0.0  
2                              0.0                 0.0                0.0  
3                              0.0                 0.0                0.0  
4                              0.0                 0.0                0.0  

[5 rows x 60 columns]

In [16]:
encoder2 = OneHotEncoder(sparse=False)
data[encoder2.get_feature_names_out()] = encoder2.fit_transform(data[['technology_name']])

In [18]:
data.head()

name                website  \
0  Cardiomo Care, Inc.  https://cardiomo.com/   
1  Cardiomo Care, Inc.  https://cardiomo.com/   
2  Cardiomo Care, Inc.  https://cardiomo.com/   
3  Cardiomo Care, Inc.  https://cardiomo.com/   
4  Cardiomo Care, Inc.  https://cardiomo.com/   

                                   short_description ipo_status  founded_on  \
0  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
1  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
2  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
3  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
4  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   

  went_public_on exited_on  num_funding_rounds last_equity_funding_type  \
0            NaN       NaN                  10                     seed   
1            NaN       NaN                  10                     seed   
2            NaN       NaN                  10                     seed   
3            NaN       NaN                  10                     seed   
4            NaN       NaN                  10                     seed   

                           last_equity_funding_total  ... industry_name_nan  \
0  {"currency":"USD","amount":10000000,"amountUSD...  ...               0.0   
1  {"currency":"USD","amount":10000000,"amountUSD...  ...               0.0   
2  {"currency":"USD","amount":10000000,"amountUSD...  ...               0.0   
3  {"currency":"USD","amount":10000000,"amountUSD...  ...               0.0   
4  {"currency":"USD","amount":10000000,"amountUSD...  ...               0.0   

  technology_name_AR and VR technology_name_Artificial Intelligence  \
0                       0.0                                     0.0   
1                       0.0                                     0.0   
2                       0.0                                     0.0   
3                       0.0                                     0.0   
4                       0.0                                     0.0   

   technology_name_Biotechnology technology_name_BlockChain  \
0                            0.0                        0.0   
1                            0.0                        0.0   
2                            0.0                        0.0   
3                            0.0                        0.0   
4                            0.0                        0.0   

  technology_name_Hardware technology_name_Science and Engineering  \
0                      0.0                                     0.0   
1                      0.0                                     0.0   
2                      0.0                                     0.0   
3                      0.0                                     0.0   
4                      0.0                                     0.0   

  technology_name_Software technology_name_Sustainability  technology_name_nan  
0                      1.0                            0.0                  0.0  
1                      1.0                            0.0                  0.0  
2                      1.0                            0.0                  0.0  
3                      1.0                            0.0                  0.0  
4                      1.0                            0.0                  0.0  

[5 rows x 69 columns]

In [14]:
data.columns

Index(['name', 'website', 'short_description', 'ipo_status', 'founded_on',
       'went_public_on', 'exited_on', 'num_funding_rounds',
       'last_equity_funding_type', 'last_equity_funding_total',
       'last_funding_at', 'headquartersCountry', 'headquartersRegion',
       'employeeCount', 'industry_name', 'technology_name', 'announcedOn',
       'stage', 'moneyRaised', 'industry_name_Advertising',
       'industry_name_Agriculture and Farming',
       'industry_name_Clothing and Apparel',
       'industry_name_Commerce and Shopping',
       'industry_name_Community and Lifestyle',
       'industry_name_Computer Hardware', 'industry_name_Consumer Electronics',
       'industry_name_Consumer Goods', 'industry_name_Content and Publishing',
       'industry_name_Data and Analytics', 'industry_name_Design',
       'industry_name_Education', 'industry_name_Energy',
       'industry_name_Environment and Sustainability', 'industry_name_Events',
       'industry_name_Financial Services', 'i

In [ ]:
grouped = data.groupby('name')['']

In [19]:
columns_to_groupby=data.columns.to_list()-industries

TypeError: unsupported operand type(s) for -: 'list' and 'list'

# Test

In [70]:
data.columns

Index(['name', 'website', 'short_description', 'ipo_status', 'founded_on',
       'went_public_on', 'exited_on', 'num_funding_rounds',
       'last_equity_funding_type', 'last_equity_funding_total',
       'last_funding_at', 'headquartersCountry', 'headquartersRegion',
       'employeeCount', 'industry_name', 'technology_name', 'announcedOn',
       'stage', 'moneyRaised'],
      dtype='object')

In [71]:
test = data[['name', 'website', 'short_description', 'ipo_status', 'founded_on',
             'went_public_on', 'exited_on', 'num_funding_rounds',
             'last_equity_funding_type', 'last_equity_funding_total',
             'last_funding_at', 'headquartersCountry', 'headquartersRegion',
             'employeeCount', 'industry_name', 'technology_name']]

In [72]:
from sklearn.preprocessing import OneHotEncoder

#encoder1 = OneHotEncoder(sparse=False)
#test[encoder1.get_feature_names_out()] = encoder1.fit_transform(test[['industry_name']])

encoder2 = OneHotEncoder(sparse=False)
test[encoder2.get_feature_names_out()] = encoder2.fit_transform(test[['technology_name']])

/var/folders/rt/gch51dj925v29nlt01bkgkg80000gn/T/ipykernel_71457/3026858227.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test[encoder2.get_feature_names_out()] = encoder2.fit_transform(test[['technology_name']])
/var/folders/rt/gch51dj925v29nlt01bkgkg80000gn/T/ipykernel_71457/3026858227.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test[encoder2.get_feature_names_out()] = encoder2.fit_transform(test[['technology_name']])


In [73]:
test

name                website  \
0       Cardiomo Care, Inc.  https://cardiomo.com/   
1       Cardiomo Care, Inc.  https://cardiomo.com/   
2       Cardiomo Care, Inc.  https://cardiomo.com/   
3       Cardiomo Care, Inc.  https://cardiomo.com/   
4       Cardiomo Care, Inc.  https://cardiomo.com/   
...                     ...                    ...   
314481                 Maze       https://maze.co/   
314482                 Maze       https://maze.co/   
314483                 Maze       https://maze.co/   
314484                 Maze       https://maze.co/   
314485                 Maze       https://maze.co/   

                                        short_description ipo_status  \
0       AI-based Remote Patient Monitoring solution to...    private   
1       AI-based Remote Patient Monitoring solution to...    private   
2       AI-based Remote Patient Monitoring solution to...    private   
3       AI-based Remote Patient Monitoring solution to...    private   
4       AI-based Remote Patient Monitoring solution to...    private   
...                                                   ...        ...   
314481  Rapid, remote testing for agile teams, from id...    private   
314482  Rapid, remote testing for agile teams, from id...    private   
314483  Rapid, remote testing for agile teams, from id...    private   
314484  Rapid, remote testing for agile teams, from id...    private   
314485  Rapid, remote testing for agile teams, from id...    private   

        founded_on went_public_on exited_on  num_funding_rounds  \
0       2016-01-27            NaN       NaN                  10   
1       2016-01-27            NaN       NaN                  10   
2       2016-01-27            NaN       NaN                  10   
3       2016-01-27            NaN       NaN                  10   
4       2016-01-27            NaN       NaN                  10   
...            ...            ...       ...                 ...   
314481  2018-01-01            NaN       NaN                   4   
314482  2018-01-01            NaN       NaN                   4   
314483  2018-01-01            NaN       NaN                   4   
314484  2018-01-01            NaN       NaN                   4   
314485  2018-01-01            NaN       NaN                   4   

       last_equity_funding_type  \
0                          seed   
1                          seed   
2                          seed   
3                          seed   
4                          seed   
...                         ...   
314481                 series_b   
314482                 series_b   
314483                 series_b   
314484                 series_b   
314485                 series_b   

                                last_equity_funding_total  ...  \
0       {"currency":"USD","amount":10000000,"amountUSD...  ...   
1       {"currency":"USD","amount":10000000,"amountUSD...  ...   
2       {"currency":"USD","amount":10000000,"amountUSD...  ...   
3       {"currency":"USD","amount":10000000,"amountUSD...  ...   
4       {"currency":"USD","amount":10000000,"amountUSD...  ...   
...                                                   ...  ...   
314481  {"currency":"USD","amount":4000000000,"amountU...  ...   
314482  {"currency":"USD","amount":4000000000,"amountU...  ...   
314483  {"currency":"USD","amount":4000000000,"amountU...  ...   
314484  {"currency":"USD","amount":4000000000,"amountU...  ...   
314485  {"currency":"USD","amount":4000000000,"amountU...  ...   

       technology_name technology_name_AR and VR  \
0             Software                       0.0   
1             Software                       0.0   
2             Software                       0.0   
3             Software                       0.0   
4             Software                       0.0   
...                ...                       ...   
314481        Software                       0.0   
314482        Software                       0.0   
314483        Software

In [74]:
test.columns

Index(['name', 'website', 'short_description', 'ipo_status', 'founded_on',
       'went_public_on', 'exited_on', 'num_funding_rounds',
       'last_equity_funding_type', 'last_equity_funding_total',
       'last_funding_at', 'headquartersCountry', 'headquartersRegion',
       'employeeCount', 'industry_name', 'technology_name',
       'technology_name_AR and VR', 'technology_name_Artificial Intelligence',
       'technology_name_Biotechnology', 'technology_name_BlockChain',
       'technology_name_Hardware', 'technology_name_Science and Engineering',
       'technology_name_Software', 'technology_name_Sustainability',
       'technology_name_nan'],
      dtype='object')

In [75]:
grouped = test.groupby(['name'])['name', 'technology_name', 'technology_name_AR and VR',
                                             'technology_name_Artificial Intelligence',
                                             'technology_name_Biotechnology', 'technology_name_BlockChain',
                                             'technology_name_Hardware', 'technology_name_Science and Engineering',
                                             'technology_name_Software', 'technology_name_Sustainability',
                                             'technology_name_nan'].max()

/var/folders/rt/gch51dj925v29nlt01bkgkg80000gn/T/ipykernel_71457/3956415841.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  grouped = test.groupby(['name'])['name', 'technology_name', 'technology_name_AR and VR',
/var/folders/rt/gch51dj925v29nlt01bkgkg80000gn/T/ipykernel_71457/3956415841.py:6: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  'technology_name_nan'].max()


In [76]:
grouped.head(20)

name  \
name                                                             
!Creatice                                            !Creatice   
#IconSource                                        #IconSource   
&ME                                                        &ME   
&Open                                                    &Open   
&SISTERS                                              &SISTERS   
'She said' App                                  'She said' App   
'hoodHeroes                                        'hoodHeroes   
(med)24                                                (med)24   
* mastBus Laundry                            * mastBus Laundry   
+ Organic Remedies                          + Organic Remedies   
--                                                          --   
.Play                                                    .Play   
.nunc                                                    .nunc   
0-tech.com                                          0-tech.com   
01 Life Science and Technology  01 Life Science and Technology   
01Melody                                              01Melody   
0Chain                                                  0Chain   
0x                                                          0x   
0xa57B Consulting                            0xa57B Consulting   
1 Bank                                                  1 Bank   

                                technology_name_AR and VR  \
name                                                        
!Creatice                                             0.0   
#IconSource                                           0.0   
&ME                                                   0.0   
&Open                                                 0.0   
&SISTERS                                              0.0   
'She said' App                                        0.0   
'hoodHeroes                                           0.0   
(med)24                                               0.0   
* mastBus Laundry                                     0.0   
+ Organic Remedies                                    0.0   
--                                                    0.0   
.Play                                                 0.0   
.nunc                                                 0.0   
0-tech.com                                            0.0   
01 Life Science and Technology                        0.0   
01Melody                                              0.0   
0Chain                                                0.0   
0x                                                    0.0   
0xa57B Consulting                                     0.0   
1 Bank                                                0.0   

                                technology_name_Artificial Intelligence  \
name                                                                      
!Creatice                                                           0.0   
#IconSource                                                         0.0   
&ME                                                                 0.0   
&Open                                                               0.0   
&SISTERS                                                            0.0   
'She said' App                                                      0.0   
'hoodHeroes                                                         0.0   
(med)24                                                             0.0   
* mastBus Laundry                                                   0.0   
+ Organic Remedies                                                  0.0   
--                                                                  0.0   
.Play                                                               0.0   
.nunc                                                               0.0   
0-tech.com                                                          1.0   
01 Life Science and Technology                                

In [77]:
grouped.columns

Index(['name', 'technology_name_AR and VR',
       'technology_name_Artificial Intelligence',
       'technology_name_Biotechnology', 'technology_name_BlockChain',
       'technology_name_Hardware', 'technology_name_Science and Engineering',
       'technology_name_Software', 'technology_name_Sustainability',
       'technology_name_nan'],
      dtype='object')

In [54]:
encoder1 = OneHotEncoder(sparse=False)
grouped[encoder1.get_feature_names_out()] = encoder1.fit_transform(grouped[['industry_name']])

In [55]:
grouped

name  \
name               industry_name                                 
#IconSource        Commerce and Shopping           #IconSource   
                   Media and Entertainment         #IconSource   
                   Sales and Marketing             #IconSource   
                   Sports                          #IconSource   
&ME                Health Care                             &ME   
...                                                        ...   
​Masto             Manufacturing                        ​Masto   
                   Music and Audio                      ​Masto   
客湖KEHU             Commerce and Shopping                客湖KEHU   
                   Sales and Marketing                  客湖KEHU   
플랩풋볼 Plab Football Sports                   플랩풋볼 Plab Football   

                                                      industry_name  \
name               industry_name                                      
#IconSource        Commerce and Shopping      Commerce and Shopping   
                   Media and Entertainment  Media and Entertainment   
                   Sales and Marketing          Sales and Marketing   
                   Sports                                    Sports   
&ME                Health Care                          Health Care   
...                                                             ...   
​Masto             Manufacturing                      Manufacturing   
                   Music and Audio                  Music and Audio   
客湖KEHU             Commerce and Shopping      Commerce and Shopping   
                   Sales and Marketing          Sales and Marketing   
플랩풋볼 Plab Football Sports                                    Sports   

                                            technology_name_AR and VR  \
name               industry_name                                        
#IconSource        Commerce and Shopping                          0.0   
                   Media and Entertainment                        0.0   
                   Sales and Marketing                            0.0   
                   Sports                                         0.0   
&ME                Health Care                                    0.0   
...                                                               ...   
​Masto             Manufacturing                                  0.0   
                   Music and Audio                                0.0   
客湖KEHU             Commerce and Shopping                          0.0   
                   Sales and Marketing                            0.0   
플랩풋볼 Plab Football Sports                                         0.0   

                                            technology_name_Artificial Intelligence  \
name               industry_name                                                      
#IconSource        Commerce and Shopping                                        0.0   
                   Media and Entertainment                                      0.0   
                   Sales and Marketing                                          0.0   
                   Sports                                                       0.0   
&ME                Health Care                                                  0.0   
...                                                                             ...   
​Masto             Manufacturing                                                0.0   
                   Music and Audio                                              0.0   
客湖KEHU             Commerce and Shopping                                        0.0   
                   Sales and Marketing                                          0.0   
플랩풋볼 Plab Football Sports                                                       0.0   

                                            technology_name_Biotechnology  \
name               industry_name                                            
#IconSource        Commerce and Shoppin

In [58]:
grouped.reset_index(drop=True, inplace=True)
grouped

name            industry_name  \
0              #IconSource    Commerce and Shopping   
1              #IconSource  Media and Entertainment   
2              #IconSource      Sales and Marketing   
3              #IconSource                   Sports   
4                      &ME              Health Care   
...                    ...                      ...   
104771              ​Masto            Manufacturing   
104772              ​Masto          Music and Audio   
104773              客湖KEHU    Commerce and Shopping   
104774              客湖KEHU      Sales and Marketing   
104775  플랩풋볼 Plab Football                   Sports   

        technology_name_AR and VR  technology_name_Artificial Intelligence  \
0                             0.0                                      0.0   
1                             0.0                                      0.0   
2                             0.0                                      0.0   
3                             0.0                                      0.0   
4                             0.0                                      0.0   
...                           ...                                      ...   
104771                        0.0                                      0.0   
104772                        0.0                                      0.0   
104773                        0.0                                      0.0   
104774                        0.0                                      0.0   
104775                        0.0                                      0.0   

        technology_name_Biotechnology  technology_name_BlockChain  \
0                                 0.0                         0.0   
1                                 0.0                         0.0   
2                                 0.0                         0.0   
3                                 0.0                         0.0   
4                                 0.0                         0.0   
...                               ...                         ...   
104771                            0.0                         0.0   
104772                            0.0                         0.0   
104773                            0.0                         0.0   
104774                            0.0                         0.0   
104775                            0.0                         0.0   

        technology_name_Hardware  technology_name_Science and Engineering  \
0                            0.0                                      0.0   
1                            0.0                                      0.0   
2                            0.0                                      0.0   
3                            0.0                                      0.0   
4                            0.0                                      0.0   
...                          ...                                      ...   
104771                       1.0                                      0.0   
104772                       1.0                                      0.0   
104773                       0.0                                      0.0   
104774                       0.0                                      0.0   
104775                       0.0                                      0.0   

        technology_name_Software  technology_name_Sustainability  ...  \
0                            1.0                             0.0  ...   
1                            1.0                             0.0  ...   
2                            1.0                             0.0  ...   
3                            1.0                             0.0  ...   
4                            1.0                             0.0  ...   
...                          ...                             ...  ...   
104771                       0.0                             0.0  ...   
104772                       0.0                             0.0  ...   
104773       

In [59]:
grouped.columns

Index(['name', 'industry_name', 'technology_name_AR and VR',
       'technology_name_Artificial Intelligence',
       'technology_name_Biotechnology', 'technology_name_BlockChain',
       'technology_name_Hardware', 'technology_name_Science and Engineering',
       'technology_name_Software', 'technology_name_Sustainability',
       'technology_name_nan', 'industry_name_Advertising',
       'industry_name_Agriculture and Farming',
       'industry_name_Clothing and Apparel',
       'industry_name_Commerce and Shopping',
       'industry_name_Community and Lifestyle',
       'industry_name_Computer Hardware', 'industry_name_Consumer Electronics',
       'industry_name_Consumer Goods', 'industry_name_Content and Publishing',
       'industry_name_Data and Analytics', 'industry_name_Design',
       'industry_name_Education', 'industry_name_Energy',
       'industry_name_Environment and Sustainability', 'industry_name_Events',
       'industry_name_Financial Services', 'industry_name_Food 

In [60]:
grouped_2 = grouped.groupby('name')[grouped.drop(columns='industry_name').columns].max()

In [62]:
grouped_2.reset_index(drop=True, inplace=True)

In [63]:
grouped_2

name  technology_name_AR and VR  \
0             #IconSource                        0.0   
1                     &ME                        0.0   
2                   &Open                        0.0   
3                &SISTERS                        0.0   
4          'She said' App                        0.0   
...                   ...                        ...   
48871     Протеже Системс                        0.0   
48872           Сибирьэко                        0.0   
48873              ​Masto                        0.0   
48874              客湖KEHU                        0.0   
48875  플랩풋볼 Plab Football                        0.0   

       technology_name_Artificial Intelligence  technology_name_Biotechnology  \
0                                          0.0                            0.0   
1                                          0.0                            0.0   
2                                          0.0                            0.0   
3                                          0.0                            0.0   
4                                          0.0                            0.0   
...                                        ...                            ...   
48871                                      0.0                            1.0   
48872                                      0.0                            0.0   
48873                                      0.0                            0.0   
48874                                      0.0                            0.0   
48875                                      0.0                            0.0   

       technology_name_BlockChain  technology_name_Hardware  \
0                             0.0                       0.0   
1                             0.0                       0.0   
2                             0.0                       0.0   
3                             0.0                       1.0   
4                             0.0                       0.0   
...                           ...                       ...   
48871                         0.0                       0.0   
48872                         0.0                       0.0   
48873                         0.0                       1.0   
48874                         0.0                       0.0   
48875                         0.0                       0.0   

       technology_name_Science and Engineering  technology_name_Software  \
0                                          0.0                       1.0   
1                                          0.0                       1.0   
2                                          0.0                       1.0   
3                                          0.0                       0.0   
4                                          0.0                       1.0   
...                                        ...                       ...   
48871                                      0.0                       0.0   
48872                                      0.0                       1.0   
48873                                      0.0                       0.0   
48874                                      0.0                       1.0   
48875                                      0.0                       1.0   

       technology_name_Sustainability  technology_name_nan  ...  \
0                                 0.0                  0.0  ...   
1                                 0.0                  0.0  ...   
2                                 0.0                  0.0  ...   
3                                 1.0                  0.0  ...   
4                                 0.0                  0.0  ...   
...                               ...                  ...  ...   
48871                             0.0                  0.0  ...   
48872                             0.0                  0.0  ...   
48873                             0.0                  0.0  ...   
48874                             0

# Remove the duplicates through encoding

This problem will be challenging as there is not really fully duplicated rows. As we said above, it is just a result of the `LEFT JOIN` that multiplicated the rows for one company (several industries, several technologies, and several funding rounds). We have to one hot encode the latter features, which is what we will be doing in this section.

At the end of this section, we should reach a shape of `(55665, >18)`.

In [88]:
data.head()

name                website  \
0  Cardiomo Care, Inc.  https://cardiomo.com/   
1  Cardiomo Care, Inc.  https://cardiomo.com/   
2  Cardiomo Care, Inc.  https://cardiomo.com/   
3  Cardiomo Care, Inc.  https://cardiomo.com/   
4  Cardiomo Care, Inc.  https://cardiomo.com/   

                                   short_description ipo_status  founded_on  \
0  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
1  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
2  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
3  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   
4  AI-based Remote Patient Monitoring solution to...    private  2016-01-27   

  went_public_on exited_on  num_funding_rounds last_equity_funding_type  \
0            NaN       NaN                  10                     seed   
1            NaN       NaN                  10                     seed   
2            NaN       NaN                  10                     seed   
3            NaN       NaN                  10                     seed   
4            NaN       NaN                  10                     seed   

                           last_equity_funding_total last_funding_at  \
0  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
1  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
2  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
3  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   
4  {"currency":"USD","amount":10000000,"amountUSD...      2020-08-01   

  headquartersCountry headquartersRegion  employeeCount industry_name  \
0                  US           New York           31.0   Health Care   
1                  US           New York           31.0   Health Care   
2                  US           New York           31.0   Health Care   
3                  US           New York           31.0   Health Care   
4                  US           New York           31.0   Health Care   

  technology_name              announcedOn     stage  \
0        Software  2019-04-15 00:00:00.000   unknown   
1        Software  2020-08-01 00:00:00.000      seed   
2        Software  2017-04-01 00:00:00.000   unknown   
3        Software  2016-08-09 00:00:00.000  pre_seed   
4        Software  2019-02-01 00:00:00.000  pre_seed   

                                         moneyRaised  
0  {"amount": 5652200, "currency": "USD", "amount...  
1  {"amount": 10000000, "currency": "USD", "amoun...  
2  {"amount": 2500000, "currency": "USD", "amount...  
3  {"amount": 13000000, "currency": "USD", "amoun...  
4  {"amount": 12000000, "currency": "USD", "amoun...

##  `industry` encoding

In [93]:
data_ind = data[['name', 'industry_name']].drop_duplicates()
data_ind.head()

name         industry_name
0       Cardiomo Care, Inc.           Health Care
20      Cardiomo Care, Inc.     Computer Hardware
40      Cardiomo Care, Inc.  Consumer Electronics
60  UniCoin Blockchain Inc.              Payments
62  UniCoin Blockchain Inc.    Financial Services

In [94]:
from sklearn.preprocessing import OneHotEncoder

encoder = OneHotEncoder(sparse=False)
data_ind[encoder.get_feature_names_out()] = encoder.fit_transform(data_ind[['industry_name']])

In [95]:
data_ind.head()

name         industry_name  industry_name_Advertising  \
0       Cardiomo Care, Inc.           Health Care                        0.0   
20      Cardiomo Care, Inc.     Computer Hardware                        0.0   
40      Cardiomo Care, Inc.  Consumer Electronics                        0.0   
60  UniCoin Blockchain Inc.              Payments                        0.0   
62  UniCoin Blockchain Inc.    Financial Services                        0.0   

    industry_name_Agriculture and Farming  industry_name_Clothing and Apparel  \
0                                     0.0                                 0.0   
20                                    0.0                                 0.0   
40                                    0.0                                 0.0   
60                                    0.0                                 0.0   
62                                    0.0                                 0.0   

    industry_name_Commerce and Shopping  \
0                                   0.0   
20                                  0.0   
40                                  0.0   
60                                  0.0   
62                                  0.0   

    industry_name_Community and Lifestyle  industry_name_Computer Hardware  \
0                                     0.0                              0.0   
20                                    0.0                              1.0   
40                                    0.0                              0.0   
60                                    0.0                              0.0   
62                                    0.0                              0.0   

    industry_name_Consumer Electronics  industry_name_Consumer Goods  ...  \
0                                  0.0                           0.0  ...   
20                                 0.0                           0.0  ...   
40                                 1.0                           0.0  ...   
60                                 0.0                           0.0  ...   
62                                 0.0                           0.0  ...   

    industry_name_Privacy and Security  industry_name_Professional Services  \
0                                  0.0                                  0.0   
20                                 0.0                                  0.0   
40                                 0.0                                  0.0   
60                                 0.0                                  0.0   
62                                 0.0                                  0.0   

    industry_name_Real Estate and Construction  \
0                                          0.0   
20                                         0.0   
40                                         0.0   
60                                         0.0   
62                                         0.0   

    industry_name_Sales and Marketing  industry_name_Software  \
0                                 0.0                     0.0   
20                                0.0                     0.0   
40                                0.0                     0.0   
60                                0.0                     0.0   
62                                0.0                     0.0   

    industry_name_Sports  industry_name_Transportation  \
0                    0.0                           0.0   
20                   0.0                           0.0   
40                   0.0                           0.0   
60                   0.0                           0.0   
62                   0.0                           0.0   

    industry_name_Travel and Tourism  industry_name_Video  industry_name_nan  
0                                0.0                  0.0                0.0  
20                               0.0                  0.0                0.0  
40                               0.0                  0.0                0.0  
60                              

In [99]:
for column in data_ind.columns[2:]:
    data_ind.rename(columns={column: column[14:]}, inplace=True)

In [101]:
data_ind.head()

name         industry_name  Advertising  \
0       Cardiomo Care, Inc.           Health Care          0.0   
20      Cardiomo Care, Inc.     Computer Hardware          0.0   
40      Cardiomo Care, Inc.  Consumer Electronics          0.0   
60  UniCoin Blockchain Inc.              Payments          0.0   
62  UniCoin Blockchain Inc.    Financial Services          0.0   

    Agriculture and Farming  Clothing and Apparel  Commerce and Shopping  \
0                       0.0                   0.0                    0.0   
20                      0.0                   0.0                    0.0   
40                      0.0                   0.0                    0.0   
60                      0.0                   0.0                    0.0   
62                      0.0                   0.0                    0.0   

    Community and Lifestyle  Computer Hardware  Consumer Electronics  \
0                       0.0                0.0                   0.0   
20                      0.0                1.0                   0.0   
40                      0.0                0.0                   1.0   
60                      0.0                0.0                   0.0   
62                      0.0                0.0                   0.0   

    Consumer Goods  ...  Privacy and Security  Professional Services  \
0              0.0  ...                   0.0                    0.0   
20             0.0  ...                   0.0                    0.0   
40             0.0  ...                   0.0                    0.0   
60             0.0  ...                   0.0                    0.0   
62             0.0  ...                   0.0                    0.0   

    Real Estate and Construction  Sales and Marketing  Software  Sports  \
0                            0.0                  0.0       0.0     0.0   
20                           0.0                  0.0       0.0     0.0   
40                           0.0                  0.0       0.0     0.0   
60                           0.0                  0.0       0.0     0.0   
62                           0.0                  0.0       0.0     0.0   

    Transportation  Travel and Tourism  Video  nan  
0              0.0                 0.0    0.0  0.0  
20             0.0                 0.0    0.0  0.0  
40             0.0                 0.0    0.0  0.0  
60             0.0                 0.0    0.0  0.0  
62             0.0                 0.0    0.0  0.0  

[5 rows x 43 columns]

In [102]:
data_ind = data_ind.groupby('name').max()
data_ind.head()

/var/folders/rt/gch51dj925v29nlt01bkgkg80000gn/T/ipykernel_71457/3379434082.py:1: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  data_ind = data_ind.groupby('name').max()


Advertising  Agriculture and Farming  Clothing and Apparel  \
name                                                                      
!Creatice            0.0                      0.0                   0.0   
#IconSource          0.0                      0.0                   0.0   
&ME                  0.0                      0.0                   0.0   
&Open                0.0                      0.0                   0.0   
&SISTERS             0.0                      0.0                   1.0   

             Commerce and Shopping  Community and Lifestyle  \
name                                                          
!Creatice                      0.0                      0.0   
#IconSource                    1.0                      0.0   
&ME                            0.0                      0.0   
&Open                          0.0                      0.0   
&SISTERS                       1.0                      0.0   

             Computer Hardware  Consumer Electronics  Consumer Goods  \
name                                                                   
!Creatice                  0.0                   0.0             0.0   
#IconSource                0.0                   0.0             0.0   
&ME                        0.0                   0.0             0.0   
&Open                      0.0                   0.0             0.0   
&SISTERS                   1.0                   0.0             1.0   

             Content and Publishing  Data and Analytics  ...  \
name                                                     ...   
!Creatice                       0.0                 0.0  ...   
#IconSource                     0.0                 0.0  ...   
&ME                             0.0                 0.0  ...   
&Open                           1.0                 0.0  ...   
&SISTERS                        0.0                 0.0  ...   

             Privacy and Security  Professional Services  \
name                                                       
!Creatice                     0.0                    0.0   
#IconSource                   0.0                    0.0   
&ME                           0.0                    0.0   
&Open                         0.0                    0.0   
&SISTERS                      0.0                    0.0   

             Real Estate and Construction  Sales and Marketing  Software  \
name                                                                       
!Creatice                             0.0                  0.0       0.0   
#IconSource                           0.0                  1.0       0.0   
&ME                                   0.0                  0.0       0.0   
&Open                                 0.0                  1.0       0.0   
&SISTERS                              0.0                  0.0       0.0   

             Sports  Transportation  Travel and Tourism  Video  nan  
name                                                                 
!Creatice       0.0             0.0                 0.0    0.0  1.0  
#IconSource     1.0             0.0                 0.0    0.0  0.0  
&ME             1.0             0.0                 0.0    0.0  0.0  
&Open           0.0             0.0                 0.0    0.0  0.0  
&SISTERS        0.0             0.0                 0.0    0.0  0.0  

[5 rows x 41 columns]

In [112]:
data_ind.shape

(55665, 41)

##  `technology` encoding

In [103]:
data_tec = data[['name', 'technology_name']].drop_duplicates()
data_tec.head()

name technology_name
0       Cardiomo Care, Inc.        Software
10      Cardiomo Care, Inc.        Hardware
60  UniCoin Blockchain Inc.      BlockChain
64        Dutch Finance Lab        Software
66                  Polybit        Software

In [104]:
encoder2 = OneHotEncoder(sparse=False)
data_tec[encoder2.get_feature_names_out()] = encoder2.fit_transform(data_tec[['technology_name']])

In [107]:
data_tec.head()

name technology_name  technology_name_AR and VR  \
0       Cardiomo Care, Inc.        Software                        0.0   
10      Cardiomo Care, Inc.        Hardware                        0.0   
60  UniCoin Blockchain Inc.      BlockChain                        0.0   
64        Dutch Finance Lab        Software                        0.0   
66                  Polybit        Software                        0.0   

    technology_name_Artificial Intelligence  technology_name_Biotechnology  \
0                                       0.0                            0.0   
10                                      0.0                            0.0   
60                                      0.0                            0.0   
64                                      0.0                            0.0   
66                                      0.0                            0.0   

    technology_name_BlockChain  technology_name_Hardware  \
0                          0.0                       0.0   
10                         0.0                       1.0   
60                         1.0                       0.0   
64                         0.0                       0.0   
66                         0.0                       0.0   

    technology_name_Science and Engineering  technology_name_Software  \
0                                       0.0                       1.0   
10                                      0.0                       0.0   
60                                      0.0                       0.0   
64                                      0.0                       1.0   
66                                      0.0                       1.0   

    technology_name_Sustainability  technology_name_nan  
0                              0.0                  0.0  
10                             0.0                  0.0  
60                             0.0                  0.0  
64                             0.0                  0.0  
66                             0.0                  0.0

In [108]:
for column in data_tec.columns[2:]:
    data_tec.rename(columns={column: column[16:]}, inplace=True)

In [109]:
data_tec.head()

name technology_name  AR and VR  \
0       Cardiomo Care, Inc.        Software        0.0   
10      Cardiomo Care, Inc.        Hardware        0.0   
60  UniCoin Blockchain Inc.      BlockChain        0.0   
64        Dutch Finance Lab        Software        0.0   
66                  Polybit        Software        0.0   

    Artificial Intelligence  Biotechnology  BlockChain  Hardware  \
0                       0.0            0.0         0.0       0.0   
10                      0.0            0.0         0.0       1.0   
60                      0.0            0.0         1.0       0.0   
64                      0.0            0.0         0.0       0.0   
66                      0.0            0.0         0.0       0.0   

    Science and Engineering  Software  Sustainability  nan  
0                       0.0       1.0             0.0  0.0  
10                      0.0       0.0             0.0  0.0  
60                      0.0       0.0             0.0  0.0  
64                      0.0       1.0             0.0  0.0  
66                      0.0       1.0             0.0  0.0

In [110]:
data_tec = data_tec.groupby('name').max()
data_tec.head()

/var/folders/rt/gch51dj925v29nlt01bkgkg80000gn/T/ipykernel_71457/225863137.py:1: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  data_tec = data_tec.groupby('name').max()


AR and VR  Artificial Intelligence  Biotechnology  BlockChain  \
name                                                                         
!Creatice          0.0                      0.0            0.0         0.0   
#IconSource        0.0                      0.0            0.0         0.0   
&ME                0.0                      0.0            0.0         0.0   
&Open              0.0                      0.0            0.0         0.0   
&SISTERS           0.0                      0.0            0.0         0.0   

             Hardware  Science and Engineering  Software  Sustainability  nan  
name                                                                           
!Creatice         0.0                      0.0       0.0             0.0  1.0  
#IconSource       0.0                      0.0       1.0             0.0  0.0  
&ME               0.0                      0.0       1.0             0.0  0.0  
&Open             0.0                      0.0       1.0             0.0  0.0  
&SISTERS          1.0                      0.0       0.0             1.0  0.0

In [111]:
data_tec.shape

(55665, 9)

## `funding_round` encoding

In [113]:
data.columns

Index(['name', 'website', 'short_description', 'ipo_status', 'founded_on',
       'went_public_on', 'exited_on', 'num_funding_rounds',
       'last_equity_funding_type', 'last_equity_funding_total',
       'last_funding_at', 'headquartersCountry', 'headquartersRegion',
       'employeeCount', 'industry_name', 'technology_name', 'announcedOn',
       'stage', 'moneyRaised'],
      dtype='object')

In [123]:
data_fun = data[['name', 'announcedOn', 'stage', 'moneyRaised']].drop_duplicates()

In [124]:
data_fun.head()

name              announcedOn     stage  \
0  Cardiomo Care, Inc.  2019-04-15 00:00:00.000   unknown   
1  Cardiomo Care, Inc.  2020-08-01 00:00:00.000      seed   
2  Cardiomo Care, Inc.  2017-04-01 00:00:00.000   unknown   
3  Cardiomo Care, Inc.  2016-08-09 00:00:00.000  pre_seed   
4  Cardiomo Care, Inc.  2019-02-01 00:00:00.000  pre_seed   

                                         moneyRaised  
0  {"amount": 5652200, "currency": "USD", "amount...  
1  {"amount": 10000000, "currency": "USD", "amoun...  
2  {"amount": 2500000, "currency": "USD", "amount...  
3  {"amount": 13000000, "currency": "USD", "amoun...  
4  {"amount": 12000000, "currency": "USD", "amoun...

In [125]:
data_fun['announcedOn'] = pd.to_datetime(data_fun['announcedOn'])

In [126]:
data_fun.sort_values(by=['name', 'announcedOn']).head(30)

name announcedOn     stage  \
62012                        !Creatice  2018-07-09  pre_seed   
251494                     #IconSource  2021-05-18      seed   
233180                             &ME  2018-08-20      seed   
233181                             &ME  2019-03-12      seed   
233182                             &ME  2020-01-18      seed   
244142                           &Open  2017-02-27  pre_seed   
244140                           &Open  2021-05-20      seed   
244141                           &Open  2022-04-17   unknown   
244139                           &Open  2022-06-22  series_a   
308288                        &SISTERS  2019-11-20   unknown   
308289                        &SISTERS  2022-01-13   unknown   
288452                  'She said' App  2021-10-20  series_a   
165977                     'hoodHeroes  2020-11-01   unknown   
165978                     'hoodHeroes  2021-11-01  pre_seed   
165979                     'hoodHeroes  2022-03-01  pre_seed   
186362                         (med)24  2020-08-17      seed   
38330                * mastBus Laundry  2018-01-01      seed   
27539               + Organic Remedies  2020-11-30       NaN   
76582                               --  2018-03-10  pre_seed   
147151                           .Play  2017-07-01  pre_seed   
147152                           .Play  2018-01-14      seed   
40090                            .nunc  2017-05-01      seed   
28294                       0-tech.com  2016-08-29  pre_seed   
207089  01 Life Science and Technology  2017-05-11      seed   
207090  01 Life Science and Technology  2018-03-28  pre_seed   
207091  01 Life Science and Technology  2020-01-20      seed   
207092  01 Life Science and Technology  2021-04-30  series_a   
207093  01 Life Science and Technology  2021-09-23  series_a   
297821                        01Melody  2019-11-01       NaN   
176751                          0Chain  2017-07-01  pre_seed   

                                              moneyRaised  
62012   {"amount": 30200000, "currency": "USD", "amoun...  
251494  {"amount": 160000000, "currency": "USD", "amou...  
233180  {"amount": 38800000, "currency": "USD", "amoun...  
233181  {"amount": 100000000, "currency": "USD", "amou...  
233182  {"amount": 67600000, "currency": "USD", "amoun...  
244142                                                NaN  
244140  {"amount": 720000000, "currency": "USD", "amou...  
244141  {"amount": 217184000, "currency": "USD", "amou...  
244139  {"amount": 2600000000, "currency": "USD", "amo...  
308288  {"amount": 25803800, "currency": "USD", "amoun...  
308289  {"amount": 84021600, "currency": "USD", "amoun...  
288452                                                NaN  
165977                                                NaN  
165978                                                NaN  
165979                                                NaN  
186362  {"amount": 654232000, "currency": "USD", "amou...  
38330   {"amount": 4696200, "currency": "USD", "amount...  
27539   {"amount": 107500000, "currency": "USD", "amou...  
76582                                                 NaN  
147151  {"amount": 34000000, "currency": "USD", "amoun...  
147152  {"amount": 120000000, "currency": "USD", "amou...  
40090                                                 NaN  
28294   {"amount": 44922400, "currency": "USD", "amoun...  
207089                                                NaN  
207090                                                NaN  
207091                                                NaN  
207092  {"amount": 160000000, "currency": "USD", "amou...  
207093  {"amount": 160000000, "currency": "USD", "amou...  
297821  {"amount": 57234000, "currency": "USD", "amoun...  
176751                                                NaN